src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -32,35 +33,54 @@ public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, o
             try
             {
                 string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
+                bool valueIsInvalid = string.IsNullOrEmpty(value);
 
                 // Disallow numeric values for enums.
-                if (!string.IsNullOrEmpty(value) &&
-                    (char.IsDigit(value[0]) ||
-                    (value[0] == '-') ||
-                    (value[0] == '+')))
-                    throw new Exception();
+                if (!valueIsInvalid &&
+                    (char.IsDigit(value[0]) || (value[0] == '-') || (value[0] == '+')))
+                {
+                    valueIsInvalid = true;
+                }
 
-                if (value != value.Trim())
```

Comment:
I changed this. The comment said "_throw if the value has whitespace_", but it was not true. It checked whitespaces just at the beginning and the end of the value, but not in the middle. And it created new string also, which is unnecessary.

Reply:
Whitespace within the string would throw at the parse anyway, so only the ends of the string need to be checked, as parse will accept trailing and leading whitespace. This could still avoid an allocation for that by examining only the first and last characters